torch/lib/THD/master_worker/master/Master.cpp

```diff
@@ -4,29 +4,33 @@
 #include "../worker/Worker.h"
 #include "../../process_group/General.hpp"
 
-
 namespace thd {
-
 namespace master {
 
 std::unique_ptr<MasterCommandChannel> masterCommandChannel;
 
 } // namespace master
-
 } // namespace thd
 
 using namespace thd;
+using namespace thd::master;
 
 bool THDMasterWorkerInit(THDChannelType channel_type) {
   if (!THDProcessGroupInit(channel_type)) return false;
 
   if (dataChannel->getRank() > 0) {
+    /*
+     * Worker initialization. If initialization succeeds it goes into
```

Comment:
I don't quite understand and not sure how I can clarify it more. Worker does not throw exception on failure it just [silently returns](https://github.com/apaszke/pytorch-dist/blob/thd-mw-error-reporting/torch/lib/THD/master_worker/worker/Worker.cpp#L28) (what implies that it doesn't go into infinite loop). So if it returns it just [immediately hits `return false;`](https://github.com/apaszke/pytorch-dist/blob/thd-mw-error-reporting/torch/lib/THD/master_worker/master/Master.cpp#L27) what should be handled properly in some other component which triggered `THDMasterWorkerInit`.